apps/nextjs/src/app/api/webhooks/clerk/route.ts

```diff
@@ -0,0 +1,95 @@
+import type { UserJSON } from "@clerk/backend";
+import type { WebhookEvent } from "@clerk/nextjs/server";
+import { posthogAiBetaServerClient } from "@oakai/core/src/analytics/posthogAiBetaServerClient";
+import { aiLogger } from "@oakai/logger";
+import * as Sentry from "@sentry/node";
+import { headers } from "next/headers";
+import { Webhook } from "svix";
+import invariant from "tiny-invariant";
+
+const WEBHOOK_SECRET = process.env.CLERK_WEBHOOK_SECRET;
+
+const log = aiLogger("webhooks");
+
+declare global {
+  interface UserPublicMetadata {
+    labs?: {
+      isDemoUser?: boolean;
+      isOnboarded?: boolean;
+      featureFlagGroup?: string | null;
+    };
+  }
+}
+
+function getPrimaryEmail(user: UserJSON): string {
+  const primaryEmail = user.email_addresses.find(
+    (email) => email.id === user.primary_email_address_id,
+  );
+  invariant(primaryEmail);
+  return primaryEmail.email_address;
+}
+
+async function syncUserToPosthog(user: UserJSON) {
+  const featureFlagGroup = user.public_metadata.labs?.featureFlagGroup ?? "";
+  posthogAiBetaServerClient.identify({
+    distinctId: getPrimaryEmail(user),
+    properties: { featureFlagGroup },
+  });
+  await posthogAiBetaServerClient.shutdown();
```

Comment:
I think I learnt that we just need to do .flush() for this, otherwise it shuts it down for anything else that imports it? "Shutdown should only be called once per request" I think was the guidance.

Reply:
Although it looks like you're only using it here so maybe not important

Reply:
Yeah, the actual goal here is to flush the events, so that makes sense. It would be much more intuitive

Reply:
I'll change it to a flush